### ISSUE TITLE
Nice to have: redex-checks for lemmas

### ISSUE THREAD

**Issue by scott-fleischman** [cocreature/pfpl-redex#4]
Just an idea: generally speaking, it would be nice to have randomized testing (using `redex-check` for instance) for the various lemmas about the languages in PFPL.

**Comment by cocreature** [comment]
Definitely, `T` already has one for the progress lemma and I was planning to add more when I get around to it so feel free to add more too.
